### PR TITLE
search by rating and favourite and some refactor

### PR DESCRIPTION
**Pull request by sujoydatta**

### CODE REVIEW

**Comment by MuhammadTamzid** [comment]
Ok

File: app/models/book.rb [hunk 1]
```diff
@@ -14,4 +14,27 @@ class Book < ActiveRecord::Base
     reject_if: proc {|attributes| attributes[:image].blank?}
 
   scope :latest, -> {order "created_at DESC"}
+  scope :favourite, -> user_id{where "id IN (select book_id from
```
Comment (dieunb) [inline app/models/book.rb]:
```
scope :favourite, -> user_id{joins(:user_books).where user_books: {user_id: user_id, favourite: true}}
```

File: app/models/book.rb [hunk 2]
```diff
@@ -14,4 +14,27 @@ class Book < ActiveRecord::Base
...
+    user_books where user_id = #{user_id} and favourite = true)"}
+  scope :not_favourite, -> user_id{where "id NOT IN (select book_id from
+    user_books where user_id = #{user_id} and favourite = true)"}
+  
+  scope :filter_book, ->(criteria) {
```
Comment (dieunb) [inline app/models/book.rb]: `do ... end` not `{}`

File: app/models/book.rb [hunk 3]
```diff
@@ -14,4 +14,27 @@ class Book < ActiveRecord::Base
...
+      when "unfavourite"
+        not_favourite search[1].to_i
+      else
+        favourite search[1].to_i
+    end
+  }
+
+ private
+ class << self
+    def ransackable_scopes auth_object = nil
```
Comment (dieunb) [inline app/models/book.rb]: indent

File: app/models/book.rb [hunk 4]
```diff
@@ -14,6 +14,27 @@ class Book < ActiveRecord::Base
...
-  scope :favourite, -> user_id{joins(:user_books).where user_books: 
-    {user_id: user_id, favourite: true}}
+  scope :favourite, -> user_id{joins(:user_books)
+    .where user_books: {user_id: user_id, favourite: true}}
+  scope :not_favourite, -> user_id{joins(:user_books)
+    .where.not user_books: {user_id: user_id, favourite: true}}
+  
+  scope :filter_book, ->(criteria) do
+    search = criteria.split
+    case search[0]
```
Comment (dieunb) [inline app/models/book.rb]:
```
+    case search[0]
 +   when "all"
 +      Book.all
 +   when "unfavourite"
 +      not_favourite search[1].to_i
 +   else
 +      favourite search[1].to_i
```

**Pull request merged**